src/wayland/shell/xdg/handlers/surface/toplevel.rs

```diff
@@ -97,26 +97,36 @@ where
                 });
             }
             xdg_toplevel::Request::SetMaximized => {
-                let handle = make_toplevel_handle(toplevel);
-                XdgShellHandler::maximize_request(state, handle);
+                if toplevel_has_capability(toplevel, xdg_toplevel::WmCapabilities::Maximize) {
```

Comment:
So we pre-filter the requests being passed to downstream by the capabilities it sets?
Does something in the protocol demand, that we don't handle these events?

Or is this to keep <v5-clients causing *unexpected* events? Because I would say let downstream handle that instead of making it impossible.

Reply:
The protocol defines that the requests should be ignored when the capability is absent. Though this could result in a race if the capabilities change during runtime. Like on creation fullscreen is supported, but later removed.